rust/agama-lib/src/network/types.rs

```diff
@@ -24,11 +31,96 @@ impl From<SSID> for Vec<u8> {
     }
 }
 
-#[derive(Debug, PartialEq, Copy, Clone)]
+#[derive(Debug, PartialEq, Copy, Clone, Serialize, Deserialize)]
 pub enum DeviceType {
     Loopback = 0,
     Ethernet = 1,
     Wireless = 2,
+    Bond = 3,
+}
+/// Bond mode
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+pub enum BondMode {
+    #[serde(rename = "balance-rr")]
```

Comment:
Just a hint. As I'm not sure what makes the more sense/ is better to read!

In our xml parsing part we used [serde_with](https://crates.io/crates/serde_with) and [strum_macros](https://crates.io/crates/strum_macros) crates to achieve similar things.
https://github.com/jcronenberg/agama/blob/523ebe9abf4055c67952d2d50e6f1fbe734044b5/rust/agama-migrate-wicked/src/interface.rs#L167